### PR TITLE
Add delivery options for Sana

### PR DESCRIPTION
**Pull request by jknapp25**
Surfacing two delivery options for Sana Care

image

### CODE REVIEW

File: apps/patient/src/components/BrandedPharmacyCard.tsx [hunk 1]
```diff
@@ -18,14 +17,13 @@ import capsulePharmacyIdLookup from '../data/capsulePharmacyIds.json';
...
-  selectedId: string;
-  handleSelect: (id: string) => void;
+  selectedId?: string;
+  handleSelect?: (id: string) => void;
 }
 
 const PHARMACY_BRANDING = {
   [process.env.REACT_APP_AMAZON_PHARMACY_ID as string]: {
-    logo: amazonPharmacyLogo,
-    description: 'Save time. Save money. Stay healthy.'
```
Comment (marisbest2) [inline apps/patient/src/components/BrandedPharmacyCard.tsx:28]: Why'd we drop this?

File: apps/patient/src/components/BrandedPharmacyCard.tsx [hunk 2]
```diff
@@ -18,14 +17,13 @@ import capsulePharmacyIdLookup from '../data/capsulePharmacyIds.json';
 
 interface Props {
   pharmacyId: string;
-  selectedId: string;
-  handleSelect: (id: string) => void;
+  selectedId?: string;
+  handleSelect?: (id: string) => void;
```
Comment (marisbest2) [inline apps/patient/src/components/BrandedPharmacyCard.tsx:21]: Why is this optional?

File: apps/patient/src/components/BrandedPharmacyCard.tsx [hunk 3]
```diff
@@ -18,14 +17,13 @@ import capsulePharmacyIdLookup from '../data/capsulePharmacyIds.json';
 
 interface Props {
   pharmacyId: string;
-  selectedId: string;
-  handleSelect: (id: string) => void;
+  selectedId?: string;
```
Comment (marisbest2) [inline apps/patient/src/components/BrandedPharmacyCard.tsx:20]: This should be `string | undefined`. You presumably always want the parent to send it down?

File: apps/patient/src/views/Pharmacy.tsx [hunk 4]
```diff
@@ -104,6 +105,10 @@ export const Pharmacy = () => {
 
   const enableMailOrder = !isDemo && orgSettings.mailOrderNavigate;
 
+  const isCapsuleTerritory =
+    order?.address?.postalCode !== undefined && order.address.postalCode in capsuleZipcodeLookup;
```
Comment (marisbest2) [inline apps/patient/src/views/Pharmacy.tsx:109]: Generally prefer `!= null` to !== undefined`.

File: apps/patient/src/views/Pharmacy.tsx [hunk 5]
```diff
@@ -611,9 +616,18 @@ export const Pharmacy = () => {
...
-              {enableMailOrder ? (
+              {enableCourier || enableMailOrder ? (
                 <BrandedOptions
-                  options={orgSettings.mailOrderNavigateProviders ?? []}
+                  options={[
+                    ...(enableCourier && order?.address?.postalCode
+                      ? [
+                          capsuleZipcodeLookup[
+                            order.address.postalCode as keyof typeof capsuleZipcodeLookup
+                          ]?.pharmacyId
```
Comment (marisbest2) [inline apps/patient/src/views/Pharmacy.tsx:626]: If the postal code is not in the lookup, you might end up with an `undefined` in there. I dont think this is possible though?

Comment (marisbest2) [reply apps/patient/src/views/Pharmacy.tsx:626]: I would probably do something like

```ts
const capsulePharmacy = capsuleZipcodeLookup[
                            order.address.postalCode as keyof typeof capsuleZipcodeLookup
                          ]?.pharmacyId;
```
And then 
```tsx
options={[...(enableCourier && order?.address?.postalCode && capsulePharmacy ? [capsulePharmacy] : []) ...)
```

Still kinda ugly but I think a bit better

Comment (marisbest2) [reply apps/patient/src/views/Pharmacy.tsx:626]: You can also do:
```ts
const options = [
    enableCourier && order?.address?.postalCode && capsulePharmacy, 
    enableMailOrder && orgSettings.mailOrderNavigateProviders
].filter(x => x)
```

File: packages/settings/src/lib/photon.ts [hunk 6]
```diff
@@ -210,8 +210,9 @@ const organizationSettings: {
     ...defaultSettings,
     logo: 'sana_care_logo.png',
     accentColor: '#FBCC45',
-    mailOrderNavigate: false,
+    mailOrderNavigate: true,
     mailOrderNavigateProviders: [AMAZON_PHARMACY_ID],
+    enableCourierNavigate: true,
```
Comment (marisbest2) [inline packages/settings/src/lib/photon.ts:215]: Can we do this in a separate PR I think, just in case this doesnt work for some reason and we want to walk it up to photon

File: apps/patient/src/views/Status.tsx [hunk 7]
```diff
@@ -178,6 +185,9 @@ export const Status = () => {
   // Demo pharmacies are already prepared
   const pharmacyWithHours = pharmacy ? (isDemo ? pharmacy : preparePharmacy(pharmacy)) : undefined;
 
+  const isDeliveryPharmacy =
+    fulfillmentType === types.FulfillmentType.MailOrder || fulfillmentType === 'COURIER';
```
Comment (marisbest2) [inline apps/patient/src/views/Status.tsx:189]: Either use strings for both or use the enum for both

File: apps/patient/src/components/BrandedPharmacyCard.tsx (hunk 1 above)
Comment (jknapp25) [reply apps/patient/src/components/BrandedPharmacyCard.tsx:28]: just to distinguish a bit more from same day, [chat w/ jomi](https://photonhealth.slack.com/archives/C0302SQME1K/p1713388912122989?thread_ts=1713387121.069459&cid=C0302SQME1K)

File: apps/patient/src/components/BrandedPharmacyCard.tsx (hunk 2 above)
Comment (jknapp25) [reply apps/patient/src/components/BrandedPharmacyCard.tsx:21]: these aren't used on Status page

File: apps/patient/src/components/BrandedPharmacyCard.tsx (hunk 3 above)
Comment (jknapp25) [reply apps/patient/src/components/BrandedPharmacyCard.tsx:20]: status page doesn't use the selectedId stuff since the pharmacy is just presentational, so i just made it optional

File: apps/patient/src/views/Pharmacy.tsx (hunk 5 above)
Comment (jknapp25) [reply apps/patient/src/views/Pharmacy.tsx]: updated

File: apps/patient/src/components/BrandedPharmacyCard.tsx [hunk 8]
```diff
@@ -51,40 +49,24 @@ export const BrandedPharmacyCard = ({ pharmacyId, selectedId, handleSelect }: Pr
...
+    </Text>
+  ) : null;
 
   return (
     <SlideFade offsetY="60px" in={true} key={`courier-pharmacy-${pharmacyId}`}>
       <Card
         bgColor="white"
         cursor="pointer"
-        onClick={() => handleSelect(pharmacyId)}
+        onClick={() => handleSelect && handleSelect(pharmacyId)}
```
Comment (captDaylight) [inline apps/patient/src/components/BrandedPharmacyCard.tsx:63]: can also use optional chaining here 
`onClick={() => handleSelect?.(pharmacyId)}`

**Review by captDaylight — APPROVED** [review]

File: apps/patient/src/components/BrandedPharmacyCard.tsx (hunk 8 above)
Comment (marisbest2) [reply apps/patient/src/components/BrandedPharmacyCard.tsx:63]: I had the same thing here lol

File: apps/patient/src/components/BrandedPharmacyCard.tsx (hunk 1 above)
Comment (jknapp25) [reply apps/patient/src/components/BrandedPharmacyCard.tsx]: reverted to original

File: apps/patient/src/components/BrandedPharmacyCard.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (jknapp25) [reply apps/patient/src/components/BrandedPharmacyCard.tsx:64]: did optional chain
